src/wallet/wallet.cpp

```diff
@@ -812,11 +813,11 @@ bool CWallet::GetAccountPubkey(CPubKey &pubKey, std::string strAccount, bool bFo
 
 void CWallet::MarkDirty()
 {
-    {
-        LOCK(cs_wallet);
-        for (std::pair<const uint256, CWalletTx>& item : mapWallet)
-            item.second.MarkDirty();
-    }
+    MarkBalancesDirty();
+
+    LOCK(cs_wallet);
```

Comment:
I guess I should move the `LOCK()` up to the top. Because it RAII, I think it doesn't matter in this case (no extra block). The lock should take affect when we enter the function/scope (before `MarkBalancesDirty()`). But maybe I'm wrong.